app/controllers/application_controller/ci_processing.rb

```diff
@@ -792,8 +804,19 @@ def discover
       end
       @in_a_form = true
       drop_breadcrumb(:name => "#{title} Discovery", :url => "/host/discover")
+      @discover_type_selected = params[:discover_type_selected]
 
-      if request.parameters[:controller] == "ems_cloud" || params[:discover_type_ipmi].to_s == "1"
+      if request.parameters[:controller] == "ems_cloud" && params[:discover_type_selected] == ExtManagementSystem.ems_cloud_discovery_types.first
```

Comment:
instead of ExtManagementSystem.ems_cloud_discovery_types.first cant we just check for "azure"